frontend/hub/remote-registries/RemoteRegistryForm.tsx

```diff
@@ -69,6 +71,75 @@ export function CreateRemoteRegistry() {
   );
 }
 
+export function EditRemoteRegistry() {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const params = useParams<{ id: string }>();
+  const name = params.id;
+  const { data, error, refresh } = useGet<HubItemsResponse<RemoteRegistryProps>>(
+    hubAPI`/_ui/v1/execution-environments/registries/?name=${name ?? ''}`
+  );
+  const getPageUrl = useGetPageUrl();
+  if (error) return <AwxError error={error} handleRefresh={refresh} />;
+  if (!data) return <LoadingPage breadcrumbs tabs />;
+
+  const remoteRegistry = data?.data[0];
+
+  const handleRefresh = () => {
+    if (!error && !remoteRegistry) {
+      navigate(-1);
```

Comment:
Yeah, that is the case that navigate is taking care of, if the user manipulates the URL, and goes to a non existent remote registry, it would navigate back to where the URL was initially modified. But we need a general standard about that one. I will verify the usePageNavigate usage here, not sure about this one.

Reply:
This seems to be a bit more nuance when to use usePageNavigate over useNavigate, I will change in a follow up PR if necessary.